ckanext/dcatapchharvest/profiles.py

```diff
@@ -309,7 +308,9 @@ def parse_dataset(self, dataset_dict, dataset_ref):  # noqa
         ):
             value = self._object_value(dataset_ref, predicate)
             if value:
-                dataset_dict[key] = self._clean_datetime(value)
+                cleaned_value = self._clean_datetime(value)
+                if cleaned_value:
```

Comment:
there is quite a lot of if-statements going on. As the `_clean_datetime`-function produces `None` for invalid dates anyways, it look and read nicer to just throw the value into the function, and return `None` if the value does not exist. Definitely matter of taste, but maybe a little easier to read.

Reply:
What do you want me to do here? Can you explain this  further? I am also using this function for temporals: there I need to know is this value valid or not. There I can't just throw it in and get a valid property out of it. So do you want me to refactor this code in any other way? Please explain and consider also the temporals.